### DISCUSSION TITLE
TrappingSR3 and ConstrainedSR3 are the same thing?

### DISCUSSION THREAD

**Discussion by waleedaero** [dynamicslab/pysindy#286] [General]
Hello,

I wanted to ask, do TrappingSR3 and ConstrainedSR3 perform the same function. As I'm trying to work with TrappingSR3, I get an error 'module 'pysindy' has no attribute 'TrappingSR3''. I have tried updating my pysindy module as well, but the problem persists.

**Comment by akaptano** [comment]
Can you print out/report your PySINDy version?

**Reply by waleedaero** [reply]
1.7.3

**Reply by akaptano** [reply]
Are you able to use any of the other new functionality, like running examples 17 and 18? Can you verify that trapping_sr3.py lives in your sources files at pysindy/pysindy/optimizers/? Is CVXPY installed on your machine?

**Comment by waleedaero** [comment]
Examples 17 and 18 are running. tra[[ing_sr3.py lives in the source file and I installed CVSPY using conda install -c conda-forge cvxpy.
The errors persist.

**Comment by akaptano** [comment]
Could you post an image of exactly what you are doing/seeing when you try to import pysindy, cvxpy, and the TrappingSR3 optimizer? I'm going to need more information to track down the issue.

**Comment by waleedaero** [comment]
It's working now. I just updated all libraries. Thank you
